### DISCUSSION TITLE
Dynamic ITypeModule and start-up time

### DISCUSSION THREAD

**Discussion by jimitndiaye** [ChilliCream/hotchocolate#6272] [Q&A] [answered]
I'm using a custom ```ITypeModule``` that looks up data from external systems to build up parts of the schema. Despite a few stumbling blocks initially we've got this working, along with filtering/sorting/projections over this custom data source. One pain point we still have however is that the initial building up of the schema can take some time  (we have to request the entity metadata from the external system then build the schema types accordingly and it can be a lot of entities)- and currently this cost is being paid for by the user who makes the first request to the graph QL schema. 

What's the recommended way to trigger building up the schema at application start-up, rather than waiting for the first request?

**Comment by tobias-tengler** [comment, marked as answer]
This should do the trick: https://chillicream.com/docs/hotchocolate/v13/performance#startup-performance
Since the warmup is done using a hosted service, the app won't accept any requests until the warmup process is finished. You can nicely combine this with readiness probes in Kubernetes for example to only route requests to the new container, once the warmup is finished.

**Reply by jimitndiaye** [reply]
Thanks. Exactly what I was looking for.
